src/Tags/Concerns/RendersForms.php

```diff
@@ -127,9 +127,15 @@ protected function getRenderableField($field, $errorBag = 'default', $manipulate
     {
         $errors = session('errors') ? session('errors')->getBag($errorBag) : new MessageBag;
 
+        $missing = str_random();
+        $old = old($field->handle(), $missing);
+        $default = $field->value() ?? $field->defaultValue();
+        $value = $old === $missing ? $default : $old;
+
```

Comment:
Context for future us, or any onlookers:

This weirdness with the random string `old` fallback is because of how Laravel's handling of old values works.

If a field is submitted with an empty value, it gets converted to `null`, so `old('fieldname', 'fallback')` will give you `null`. If the field wasn't submitted at all, the `fallback` would be returned.

If you were to intentionally submit a text field with an empty value, there's no way to tell the difference between the old empty value, or a completely missing value. If we did `old('field') ?? $default` then we'd end up rendering the default value when a user empties out the field - that's not right.

One option would be to exclude that field from the `ConvertEmptyStringsToNulls` middleware, but that's overkill and we don't have control of that anyway.

This is the alternative - if the `old()` method gives us the fallback, we can be sure it's actually missing and not just emptied by the user. I'm using a random string just to prevent someone being able to submit the known string.